File: 백준/Silver/28278. 스택 2/스택 2.py
import sys

N = int(sys.stdin.readline())
stack = []
ans = []
for i in range(N) :
    j = sys.stdin.readline().split()
    
    if j[0] == '1' :
        stack.append(j[1])
    elif j[0] == '2' :
        if len(stack) > 0 :
            top = stack.pop()
            ans.append(top)
        else :
            ans.append('-1')
    elif j[0] == '3' :
        ans.append(str(len(stack)))
    elif j[0] == '4' :
        if len(stack) == 0 :
            ans.append('1')
        else :
            ans.append('0')
    else :
        if len(stack) > 0 :
            ans.append(stack[-1])
        else :
            ans.append('-1')
print("\n".join(ans))

File: 백준/Silver/28278. 스택 2/스택 2.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const [N, ...input] = fs.readFileSync(filePath).toString().split("\n");

const stack = [];
const answer = [];
for (const command of input) {
    const act = command.split(" ");
    if (act[0] == "1") {
        stack.push(act[1]);
    } else if (act[0] == "2") {
        if (stack.length > 0) {
            const pop = stack.pop();
            answer.push(pop);
        } else {
            answer.push(-1);
        }
    } else if (act[0] == "3") {
        answer.push(stack.length);
    } else if (act[0] == "4") {
        if (stack.length > 0) {
            answer.push(0);
        } else {
            answer.push(1);
        }
    } else if (act[0] == "5") {
        if (stack.length > 0) {
            answer.push(stack[stack.length - 1]);
        } else {
            answer.push(-1);
        }
    }
}

console.log(answer.join("\n"));